cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +89,410 @@ def unset_default_org(self):
+        |   |-- alias2.service
+        |   |-- ...
+        |-- devhub
+        |   |-- alias1.service
+        |   |-- alias2.service
+        |   |-- ...
+        .
+
+        This also has the advantage that when a new service type
+        is added to cumulusci.yml a new directory for that service type
+        will be created.
+        """
+        services_dir_path = Path(f"{self.global_config_dir}/services")
+        services_dir_path.mkdir(exist_ok=True)
+
+        configured_service_types = self.project_config.config["services"].keys()
+        for service_type in configured_service_types:
+            service_type_dir_path = Path(services_dir_path / service_type)
+            if not Path.is_dir(service_type_dir_path):
+                Path.mkdir(service_type_dir_path)
+
+    def _migrate_services(self):
+        """Migrate .service files from the global_config_dir and
+        any project local directories."""
+        self._migrate_services_from_dir(self.global_config_dir)
+        for local_proj_dir in self._iter_local_project_dirs():
+            self._migrate_services_from_dir(local_proj_dir)
+
+    def _migrate_services_from_dir(self, dir_path: str) -> None:
+        """Migrate all .service files from the given directory to
+        the appropriate service sub-directory and apply the default
+        alias. This is intended to be run against either the
+        global_config_dir, or a local project directory.
+
+        Default aliases are in the form `service_type__scope`.
+        Scope is either the name of the local project directory
+        or 'global'; depending on where the .service file is located."""
+        dir_path = Path(dir_path)
+        for item in dir_path.iterdir():
+            if item.suffix == ".service":
+                new_service_filepath = self._get_new_service_filepath(item)
+                if not new_service_filepath:
+                    continue
+                if new_service_filepath.is_file():
+                    self.logger.warning(
+                        f"Skipping migration of {item.name} as a default alias already exists for this service type. "
+                    )
+                    continue
+
+                item.replace(new_service_filepath)
+
+    def _get_new_service_filepath(self, item: Path) -> Path:
+        """Given an old .service filepath, determine the path
+        and filename of the new .service file.
+
+        @returns: the Path to the newfile, or None if the service
+        is of an unrecognized type.
+        """
+        service_type = item.name.replace(".service", "")
+        configured_service_types = self.project_config.config["services"].keys()
+        if service_type not in configured_service_types:
+            self.logger.info(f"Skipping migration of unrecognized service: {item.name}")
+            return None
+
+        scope = "global" if item.parent.name == ".cumulusci" else "project"
+        new_filename = f"{service_type}__{scope}.service"
```

Comment:
Should this just be f"{scope}.service"? I was expecting to have a github service named `global` after the migration, but instead I have one named `github__global`:

<img width="272" alt="Screen Shot 2021-04-09 at 11 02 58 AM" src="https://user-images.githubusercontent.com/49014/114200695-745ba400-9923-11eb-90c9-07a93f7ef9b2.png">

Reply:
We did intend to have the names like this but I think your suggestion is better.